drivers/SMBSR.py

```diff
@@ -195,9 +195,17 @@ def attach(self, sr_uuid):
             try:
                 self.mount()
                 os.symlink(self.linkpath, self.path)
+                self._check_writable()
+                self._check_hardlinks()
             except SMBException as exc:
                 raise xs_errors.XenError('SMBMount', opterr=exc.errstr)
-            self._check_hardlinks()
+            except:
+                if util.pathexists(self.path):
+                    os.unlink(self.path)
+                if self.checkmount():
```

Comment:
errr, you need to unmount before removing the mountpoint.

Reply:
`self.path` is just a symlink, created after mounting. The mountpoint is deleted by `unmount`.